### PR TITLE
upgrade: Kubeflow 1.4.0-rc.0 upgrade structure adjustment. Partial #302

### PR DESCRIPTION
**Pull request by zijianjoy**
Partial #302

Change:

1. tf-training and pytorch-job are merged into training-operator.
2. Depends on kubeflow 1.4.0-rc.0, but we are expecting rc.1 to show up soon.
3. Depends on KFP 1.7.0 which is different from kubeflow rc.0 for now.

### CODE REVIEW

File: kubeflow/apps/pipelines/pull-upstream.sh [hunk 1]
```diff
@@ -17,10 +17,10 @@
...
 
 # TODO(Bobgy): use KFP 1.6.1 when https://github.com/kubeflow/pipelines/pull/5750 is released.
-# export KUBEFLOW_PIPELINES_VERSION=1.5.0
-# export KUBEFLOW_PIPELINES_REPO=https://github.com/kubeflow/pipelines.git
-export KUBEFLOW_PIPELINES_VERSION=config-default-pipeline-root
-export KUBEFLOW_PIPELINES_REPO=https://github.com/Bobgy/pipelines.git
+export KUBEFLOW_PIPELINES_VERSION=1.7.0
+export KUBEFLOW_PIPELINES_REPO=https://github.com/kubeflow/pipelines.git
+# export KUBEFLOW_PIPELINES_VERSION=config-default-pipeline-root
+# export KUBEFLOW_PIPELINES_REPO=https://github.com/Bobgy/pipelines.git
```
Comment (Bobgy) [inline kubeflow/apps/pipelines/pull-upstream.sh:23]: remove the reference to my fork repo?

File: kubeflow/config.yaml [hunk 2]
```diff
@@ -95,15 +95,14 @@ components:
...
-
-# Tensorflow operator
-# https://www.kubeflow.org/docs/components/training/tftraining/
-# dependencies: []
-- apps/tf-training
+- apps/training-operator
+    # Tensorflow operator
+    # https://www.kubeflow.org/docs/components/training/tftraining/
+    # dependencies: []
+    # - apps/tf-training
```
Comment (Bobgy) [inline kubeflow/config.yaml:105]: update the comment?

**Review by Bobgy — COMMENTED** [review]
/lgtm

File: kubeflow/apps/pipelines/pull-upstream.sh (hunk 1 above)
Comment (zijianjoy) [reply kubeflow/apps/pipelines/pull-upstream.sh:23]: Sounds good, will make this change in next PR.

File: kubeflow/config.yaml (hunk 2 above)
Comment (zijianjoy) [reply kubeflow/config.yaml:105]: Sounds good, will update in next PR.

**Pull request merged**